Migrate FDM to JSBSim

BobDotCom opened the pull request.

BobDotCom committed:
feat: initial jsbsim commit

BobDotCom committed:
feat: improve fdm

BobDotCom committed:
feat: fix htail

BobDotCom committed:
feat: move tail closer

BobDotCom committed:
feat: add web aero

BobDotCom committed:
fix: increase max thrust & new augmethod

BobDotCom committed:
fix: use engine file

BobDotCom committed:
fix: fix afterburner

BobDotCom committed:
fix: fix engine indicators

BobDotCom committed:
feat: apply eng ind changes

BobDotCom committed:
Merge branch 'merge-1' into jsbsim

BobDotCom committed:
feat: new engine panel

BobDotCom committed:
style: reformat eng_ind

BobDotCom committed:
fix: use celsius for eng temp indicator

BobDotCom committed:
fix: shift image to original position

BobDotCom committed:
fix: fix engine start panel texture

BobDotCom committed:
Merge branch 'master' into jsbsim

# Conflicts:
#	Models/Instruments-3d/cp.png
#	Models/Instruments-3d/eng_ind.ac
#	Models/Instruments-3d/eng_ind.xml

BobDotCom committed:
Merge branch 'master' into jsbsim

BobDotCom committed:
Merge branch 'master' into jsbsim

BobDotCom committed:
feat: basic engine systems

BobDotCom committed:
feat: basic pneumatic systems

BobDotCom committed:
feat: add adg to engine system

BobDotCom committed:
Merge branch 'master' into jsbsim

BobDotCom committed:
Merge branch 'master' into jsbsim

BobDotCom committed:
fix: fix engine switch panel

BobDotCom committed:
feat: APU RUN indicators

BobDotCom committed:
feat: APU master switch snap-back

BobDotCom committed:
feat: basic electrical system

BobDotCom committed:
feat: startup checklists

BobDotCom committed:
fix: regulate ac at 230v

BobDotCom committed:
chore: add a project config file

BobDotCom committed:
Merge branch 'master' into jsbsim

BobDotCom committed:
refactor: rename dc 1/2 to dc fwd/aft

BobDotCom committed:
refactor: rename dc hot busses

BobDotCom committed:
refactor: rename idg gcu

BobDotCom committed:
fix: dont trip gcu if adg decoupled

BobDotCom committed:
feat: use electrical for engine start

BobDotCom committed:
fix: engine start require both elec and drive

BobDotCom committed:
feat: als afterburners

BobDotCom committed:
feat: engine nozzles use real prop

BobDotCom committed:
Merge branch 'master' into jsbsim

BobDotCom committed:
feat: start new FDM

BobDotCom committed:
feat: new generated file

BobDotCom committed:
Merge branch 'master' into jsbsim

# Conflicts:
#	Models/B-1B.xml
#	Models/Instruments-3d/overhead_panel.ac

BobDotCom committed:
Merge branch 'master' into jsbsim

BobDotCom committed:
chore: delete old run script

BobDotCom committed:
chore: some new experiments

BobDotCom committed:
chore: input file

BobDotCom committed:
feat: move landing gear

BobDotCom committed:
feat: add radome contact

BobDotCom committed:
feat: fdm tuner dialogs

BobDotCom committed:
fix: fix fdm Drag_beta tuner

BobDotCom committed:
feat: switch to elevons & remove ailerons

BobDotCom committed:
feat: jsbsim wing sweep

BobDotCom committed:
fix: add missing file

BobDotCom committed:
feat: add raw aero fdm

RobAndrew96 committed:
Update b1b-electrical.xml

Added power to nav and adf sources

BobDotCom committed:
Merge branch 'jsbsim' of https://github.com/BobDotCom/B-1B into jsbsim

BobDotCom committed:
feat: add correct fuel tanks

BobDotCom committed:
fix: fix autopilot speed modes

BobDotCom committed:
feat: wing sweep in fdm

BobDotCom committed:
feat: temp save

BobDotCom committed:
fix(fdm): reverse wing-sweep-deg

BobDotCom committed:
feat(fdm): use concorde data

BobDotCom committed:
feat(fdm): add sweep tuner dial

BobDotCom committed:
fix(fdm): tune alpha lift

BobDotCom committed:
fix(fdm): tune flap drag

BobDotCom committed:
chore: push some files i had lying around

BobDotCom committed:
fix: re-number electrical generators

BobDotCom committed:
fix: re-number electrical generators pt2

BobDotCom committed:
fix: re-number IDG comments

BobDotCom committed:
feat(elec): add idg 1

BobDotCom committed:
fix(elec): use correct generators for ac busses

BobDotCom committed:
fix(elec): idg use correct engine fire button

BobDotCom committed:
feat(elec): add ac bus 3+4

BobDotCom committed:
Merge branch 'master' into jsbsim

BobDotCom committed:
refactor(elec): use int for generator switches

BobDotCom committed:
feat(engines): use correct busses for startup

RobAndrew96 committed:
New Overhead Panel docs for generator switches

RobAndrew96 committed:
BIG PUSH! Lot's of things added, mainly: 
Lighting to instruments and lighting controls on overhead 
Some of Comm adjusters on Captains side now functional 
IFF panel dials adjustable but not functional yet 
Fuel flow dial is now functional and correct 
ADC and INS switches on Captains side panel functional. Failures not implemented 
FLT-CTRL-PWR panel AUG switches work to some degree, albeit wrongly, this is not what they do. This will be re-written soon 
New Annunciator panels added. Indicators being worked on, showing constantly at the moment. 

    I think most changes have been added. Will be checking over to make sure everything is moved across.

RobAndrew96 committed:
A few more changes from the last push

RobAndrew96 committed:
Some graphical changes as well as TTO Function enabled

RobAndrew96 committed:
Really need to upload more often...
Bunch of things added. Bearing needle on HSI now points to target when using waypoints in route manager. HSI also gives distances for individual waypoints (both when Fltdir set to OFF). additional external position lights added to rear of wingtips (ALS only) as well as refuelling lighting added with lightmap. Oooh how fancy, I know. Much more to come. Hopefully more frequently to. this is to dangerous!

RobAndrew96 committed:
Additional stuff. Last push had error

RobAndrew96 committed:
Small fixes

RobAndrew96 committed:
Added functionality to IFF panel (Yet to test) as well as the fltdir panel. Not perfect but certainly better and more functional than before.

RobAndrew96 committed:
Additional graphical corrections

RobAndrew96 committed:
Added new engine and APU sounds. Also adjusted view positions and aircraft position. As well as a few in cockpit fixes

RobAndrew96 committed:
Adding correction to HSI Indicator

RobAndrew96 committed:
Some more changes made to models

RobAndrew96 committed:
Added unfinished hydraulic system to aircraft.

BobDotCom committed:
chore: move unused aero files into directory

RobAndrew96 committed:
Moved aero files to jsbsim/aero

RobAndrew96 committed:
Removed useless animations in Annunciator panel and arms panel

BobDotCom committed:
Merge branch 'master' into jsbsim

# Conflicts:
#	Models/Instruments-3d/IFF-Panel.ac
#	Models/Instruments-3d/test.png

BobDotCom commented on Systems/computers.xml:
```diff
@@ -0,0 +1,573 @@
+        </condition>
+       </enable>
+       <input>
+       <property>orientation/roll-deg</property> 
```
Maybe this should be instrumentation/attitude-indicator/indicated-roll-deg instead?

RobAndrew96 replied:
Yeah. If we REALLY wanted to dive in to this, we'd simulate every single computer and have properties that did the calculations and spat out an output (I guess some things such as autopilot is in fact this). However the way it's working at the moment is that, those 8 or however many computers, are all simulated by the one each of us are sat in front of! But yes, I think this should change. I have already started the process with the ADC's and GSS's

BobDotCom committed:
Merge branch 'master' into jsbsim

# Conflicts:
#	Models/Instruments-3d/pfdb1_left.xml
#	b1-base.xml

BobDotCom commented on b1-base.xml:
```diff
@@ -439,18 +452,70 @@ B-1B set file.
+            <property-rule n="200">
+                <path>Systems/annunciators.xml</path>
+            </property-rule>
+            <property-rule n="200">
```
does this need to be a different number than the annunciators?

RobAndrew96 replied:
I'm not actually to sure what that number represents. However I do know that the file is being called upon and is working, so the fact that it's the same number as computers.xml isn't an issue I don't think? When you say "Different number than the annunciators" Do you mean the amount of annunciators?

BobDotCom replied:
Basically, each definition should be assigned a number, i.e. just above this line you have autopilot n=0 and n=1. These numbers should be unique for each definition.

BobDotCom committed:
Merge branch 'master' into jsbsim

RobAndrew96 committed:
New code put in for TO/LDG Taxi steering switch, which should now be operational (Limits sweep to 30 deg either way) Also veiw in rear seat changed

RobAndrew96 committed:
Rear seat view change

RobAndrew96 committed:
Added new Rotary weapon launchers to all weapons bays

RobAndrew96 committed:
So, I did a thing and added a new splash screen. I think you'll like it. I want to add more in fuure but need to get similar shots

RobAndrew96 committed:
ADG's now modelled correctly (mostly) Will fine tune once Hydraulic systems in place.

RobAndrew96 committed:
Finally moved 4 hydraulic systems over. Nothing connected yet other than gauges. A few other small bells and whisles, startup procedure changed as well as checklist.

RobAndrew96 committed:
A small-ish update regarding Engine fires. The extinguishing system now exists and that's about it. There's a bunch more work that needs doing regarding flame effects and the effects of fire. Modelling is done though.

RobAndrew96 committed:
Added more functionality to Fire systems such as Fire loops and loop lights. And some lighting things.

RobAndrew96 committed:
Made a start on Electric connections, starting with Overhead panel indicator knob, Nuc panel and Pump panel

RobAndrew96 committed:
Added functional failures, both pump failure and line rupture

RobAndrew96 committed:
More electrical ties, specifically to Landing gear indicator lights, Annunciator main panel and Autopilot panel

RobAndrew96 committed:
Moving Afterburner from Alter throttle switches to Throttle Quadrant

RobAndrew96 committed:
More electrical edits made. Annunciator panels should be working correctly showing correct faults. Gear moving light added

RobAndrew96 committed:
A few small bug fixes

RobAndrew96 committed:
The ones tat got away!

RobAndrew96 committed:
Some graphical/switches changes to SMS panel, and some un-finished work with control surfaces, including bay doors

RobAndrew96 committed:
Added functionality to flaps, almost finished, needs some more refining for G-loads. Annunciator panel indicators added

SammySkycrafts committed:
Code: Remove empty tests causing sim crash in some cases

SammySkycrafts committed:
Sim: Start removing old file references

RobAndrew96 committed:
Adding Launcher texture

SammySkycrafts committed:
3D: Update Clock texture

SammySkycrafts committed:
Sim: Fix or Remove broken animation objects

RobAndrew96 committed:
Added new Refuelling recepticle lights

RobAndrew96 committed:
Added new IFF panel model and programming. Also new instrumentation.xml in systems and SMS panel

RobAndrew96 committed:
Edited gaphical error with rear stabs, and some fcs based changes. Also flap G-out

SammySkycrafts committed:
FDM: Fix invalid JSBSim

RobAndrew96 committed:
Added new overhead panel textures and corrected switches. Also corrected/ing screen data

RobAndrew96 committed:
Merge branch 'jsbsim' of https://github.com/BobDotCom/B-1B into jsbsim

RobAndrew96 committed:
Small change made to HSI

RobAndrew96 committed:
Small amount added to checklists

RobAndrew96 committed:
Updating a few things relating to Pilot's screen

RobAndrew96 committed:
Moved some cockpit models around including CDU900

RobAndrew96 committed:
Added On Off switch and Test capability for TACAN

RobAndrew96 committed:
Added animation and fixes to switches on panels FLT CNTR PWR, ENG AUX PWR and TO LDG. No functionality yet.

SammySkycrafts committed:
Sim: Fix Instrumentation XML not defined

RobAndrew96 committed:
Moved some instruments around on the front wall to make space for new Alt Mach tapes

RobAndrew96 committed:
Made some model changes to newer front MFD's as well as graphic changes to borders

SammySkycrafts committed:
Fix Block E MFD Position

RobAndrew96 committed:
Added some changes in regards to centre console switches. Some now move with no function. just to set examples to use for further development. Plus some sound editing

RobAndrew96 committed:
Replacing "annunciators.png" in preperation for electrical conponents replacement

SammySkycrafts committed:
Sim: Use default FG Autopilot
This may not work good on 2020.3.x

SammySkycrafts committed:
Arms: Add JSB PointMasses for bays

RobAndrew96 committed:
Added new custom electrical system. 90% done just a few small details to add. Also added new Accumulators for start up.

RobAndrew96 committed:
Small fixes made to annunciator lighting

RobAndrew96 committed:
More fixes towards power sources. Flap functionality also improved

RobAndrew96 committed:
Added new functionality/realism to wing sweep mechinism, as well as flashing state for annunciator lights. Yet to add failures/alternate wingsweep

RobAndrew96 committed:
Feature/displays_JSBsim_Merge (#54)

* Sim: Split displays.nas to specific Block-E file, stop calling as module

* 3D: VSD prep for Canvas

* Sim: Add DU Font by Octal450

* VSD: Add WIP Canvas VSD for block d

* VSD: Remove some debug statements

* Sim: Only draw cockpit instruments when <30m to increase MP Performance

* VSD: Add roll ticks at 10,20,30 and 45 degrees

* VSD: Update hdg tape

* VSD: Add ILS/GS
Flight Director switch needs to be in TEST or ILS position to work

AND THANKS JOSH...

* VSD: Fix 2 props causing issues

* FX: ADD CRT EFFECT

* Add missing file

* Added Brightness function to screen knob. Didn't touch contrast as the effect seemed weird and didn't seem to contrast what needed contrasting

* Sim: Fix Instrumentation XML not defined

* FX: Update CRT Effect

* Merge JSBSim

* PFD: Initial HSD Work

* Block-E: Refactor display SVG Parsing

* Added PFR controls and thier location. Currently both plt/Co-plt in one model, this may change in future

---------

Co-authored-by: SammySkycrafts <ttrmine@gmail.com>

RobAndrew96 committed:
A few small changes to add before adding bigger changes later

RobAndrew96 committed:
Electrical changes for canvas based screens

RobAndrew96 committed:
Changed output name of battery busses to "voltage"

SammySkycrafts committed:
rob no. you can do that!

SammySkycrafts committed:
PFD: Fix Screens
Block E not tested, expect something to be broken

RobAndrew96 committed:
Added foundations of flight data computers, lots more to do with this.

RobAndrew96 committed:
Adding new Altitude/Mach displays. A little more work to do on them, but 80% finished

RobAndrew96 committed:
Some animation changes to the HSI

RobAndrew96 committed:
Made a start on the Radar Altimeter.

RobAndrew96 committed:
Lighting changes made to several models.

RobAndrew96 committed:
New Radar altimeter data input made in instrumentation, implemented through computers.xml. Fixes made to HSI.

RobAndrew96 committed:
Added some more detail to various checklists

SammySkycrafts committed:
Sim: Remove old missing nasal links

SammySkycrafts committed:
Code: Fix broken links in sms.xml

SammySkycrafts committed:
Code: Fix invalid instrumentation XML

SammySkycrafts committed:
Code: Cleanup

SammySkycrafts committed:
Sound: Fix sounds with no condition

SammySkycrafts committed:
Code: Fix hud_alt.xml

RobAndrew96 committed:
Small interface change to ILS knob annimation. Small changes made to base.xml

RobAndrew96 committed:
Some changes made to reported errors in aircraft

RobAndrew96 committed:
Changes made to knobs on screen in preperation for adding functionality.

RobAndrew96 committed:
Small temporary revision made to elevons

RobAndrew96 committed:
New autopilot controls and features implemented. Due to test branch for changes.

RobAndrew96 committed:
Removed Autopilot link from base.xml

RobAndrew96 committed:
Re-adding link to "Systems/Libraries" in Base.xml

RobAndrew96 committed:
Merge branch 'jsbsim' of https://github.com/BobDotCom/B-1B into jsbsim

RobAndrew96 committed:
Added graphical change to flt_dir and name change to item in PFR-set

RobAndrew96 committed:
Made animation changes to rear end turbine blades

RobAndrew96 committed:
Many changes made to fire detector loops/extinguishers for main engines only. More work to be done towards APU's.

RobAndrew96 committed:
Small change made to top annunciator panel

SammySkycrafts committed:
Sim: Actually init hackCanvas

BobDotCom committed:
Merge branch 'master' into jsbsim

RobAndrew96 committed:
Big changes made to input sources. New Pitot and static sources added and linked to new instrumentation in turn linked to CADCs.

RobAndrew96 committed:
Removing some unnesessary files

RobAndrew96 committed:
HSI outer needle now points towards TACAN source in correct setting. ILS panel re-modelling

RobAndrew96 committed:
Radar Altimeter implemented properly. Both channels working with failures.

RobAndrew96 committed:
Graphical changes to IFF and Pump panel (dump switch cover)

RobAndrew96 committed:
Made improvements to Transponder (It actually works now) and some bug fixes

RobAndrew96 committed:
Removed a ton of verticies from various switch panels. More to come hopefully!

RobAndrew96 committed:
Removing errors such as sound file config, re-introducing failures.nas and removing label text from "fire_ind"

RobAndrew96 committed:
Moved a lot of the rear seat instruments to thier correct positions. Gave "102knts" a retexture.
